### PR TITLE
[MM-43133] Open modal for widgets

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
On clicking the widget header it will open the modal, right now I have only added the top reactions data table.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43133

#### Related Pull Requests
- This needs to merge first -> https://github.com/mattermost/mattermost-webapp/pull/10147

#### Screenshots
Screen Shot 2022-04-11 at 12 38 14 PM

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/activity_and_insights/insights/time_frame_dropdown/time_frame_dropdown.tsx [hunk 1]
```diff
@@ -35,6 +37,10 @@ const TimeFrameDropdown = (props: Props) => {
             ...provided,
             cursor: 'pointer',
         }),
+        menuPortal: (provided: React.CSSProperties) => ({
+            ...provided,
+            zIndex: 100000,
```
Comment (mkraft) [inline components/activity_and_insights/insights/time_frame_dropdown/time_frame_dropdown.tsx:42]: Does it need to be this high?

File: components/activity_and_insights/insights/top_reactions/top_reactions_table/top_reactions_table.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                        </span>
+                    ),
+                    reaction: (
+                        <>
+                            <RenderEmoji
+                                emojiName={'grinning'}
+                                size={24}
+                            />
+                            <span className='cell-text'>
+                                {'grinning'}
```
Comment (mkraft) [inline components/activity_and_insights/insights/top_reactions/top_reactions_table/top_reactions_table.tsx:64]: These are hard-coded temporarily until the data wire-up?

File: components/activity_and_insights/insights/top_reactions/top_reactions.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                <CircleLoader
+                    size={20}
+                />
+            </div>
+            <div className='bar-chart-entry'>
+                <BarChartLoader/>
+                <CircleLoader
+                    size={20}
+                />
+            </div>
```
Comment (mkraft) [inline components/activity_and_insights/insights/top_reactions/top_reactions.tsx:44]: Is this temporary until the data-wire up?

**Review by mkraft — COMMENTED** [review]
Looks good. What were you planning in terms of testing?

File: components/activity_and_insights/insights/top_reactions/top_reactions_table/top_reactions_table.tsx (hunk 2 above)
Comment (BenCookie95) [reply components/activity_and_insights/insights/top_reactions/top_reactions_table/top_reactions_table.tsx:64]: Yep

File: components/activity_and_insights/insights/top_reactions/top_reactions.tsx (hunk 3 above)
Comment (BenCookie95) [reply components/activity_and_insights/insights/top_reactions/top_reactions.tsx:44]: Yeah I'll fix up this part when the data is in

**Comment by BenCookie95** [comment]
> Looks good. What were you planning in terms of testing?

For testing it mostly just needs an eye from Matt to ensure the styles are good. In terms of functional QA, just ensure the right modal titles/subtitles are there when `My insights` or `Team insights` is selected

**Review by mylonsuren — APPROVED** [review]

File: components/activity_and_insights/insights/time_frame_dropdown/time_frame_dropdown.tsx (hunk 1 above)
Comment (BenCookie95) [reply components/activity_and_insights/insights/time_frame_dropdown/time_frame_dropdown.tsx]: I've reduced it to 1100, seems to be the lowest I can use on the modal

**Review by mkraft — APPROVED** [review]

**Comment by matthewbirtch** [comment]
@BenCookie95 I don't see the insights section on the cloud test server I assume we need the feature flag turned on? @furqanmlk could you help out?

**Comment by furqanmlk** [comment]
> @BenCookie95 I don't see the insights section on the cloud test server, so I assume we need the feature flag turned on? @furqanmlk could you help out?

@matthewbirtch 
Done

https://mattermost-webapp-pr-10164.test.mattermost.cloud/pr10164/activity-and-insights

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Looks really good @BenCookie95. Nice work! Just a few things from me:

1. Can we center the modal vertical on the screen? (we're moving towards this for all modals) - see the "Create new channel" modal for an example
2. Looks like there are some styling issues with the table in other themes:
image

3. Similar styling issue with the time period select input
image

Let me know if you need specific values on colors.

**Comment by BenCookie95** [comment]
> Looks really good @BenCookie95. Nice work! Just a few things from me:
> 
> 1. Can we center the modal vertical on the screen? (we're moving towards this for all modals) - see the "Create new channel" modal for an example
> 2. Looks like there are some styling issues with the table in other themes:
> 
> image
> 
> 3. Similar styling issue with the time period select input
> 
> image
> 
> Let me know if you need specific values on colors.

I'll take a look into the styling issues, unsure what the values should be right now but I'll reach out if I'm not able to find them

**Comment by BenCookie95** [comment]
> Looks really good @BenCookie95. Nice work! Just a few things from me:
> 
> 1. Can we center the modal vertical on the screen? (we're moving towards this for all modals) - see the "Create new channel" modal for an example
> 2. Looks like there are some styling issues with the table in other themes:
> 
> image
> 
> 3. Similar styling issue with the time period select input
> 
> image
> 
> Let me know if you need specific values on colors.

@matthewbirtch those issues are fixed

**Review by mylonsuren — APPROVED** [review]

**Review by matthewbirtch — APPROVED** [review]
Looks great @BenCookie95! Thanks!

**Review by furqanmlk — COMMENTED** [review]
image

image

1. ```Time used``` is not in center
2. Sorting icon is not showing 
3. Blue bar is not showing

**Pull request merged**